File: problem_set/delete-middle-node-lcci/delete-middle-node-lcci.cpp

```
// @Title: 删除中间节点 (Delete Middle Node LCCI)
// @Author: robert.sunq
// @Date: 2020-03-25 21:36:29
// @Runtime: 12 ms
// @Memory: 6.7 MB

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    void deleteNode(ListNode* node) {
        while(node->next->next != NULL){
            node->val = node->next->val;
            node = node->next;
        }
        node->val = node->next->val;
        node->next = NULL;

    }
};

File: problem_set/delete-middle-node-lcci/delete-middle-node-lcci.java

```
// @Title: 删除中间节点 (Delete Middle Node LCCI)
// @Author: robert.sunq
// @Date: 2021-08-02 22:11:57
// @Runtime: 0 ms
// @Memory: 37.9 MB

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
        node.val = node.next.val;
        node.next = node.next.next;
    }
}
